_rules/printable-characters-shortcut-ffbc54.md

```diff
@@ -0,0 +1,483 @@
+---
+id: ffbc54
+name: No keyboard shortcuts use only printable characters
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters, then there is a mechanism to disable the shortcut, or to remap the shortcut to use one or more non-printable characters keys, or the shortcut for a user interface component is only available when that component has focus.
+accessibility_requirements:
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+acknowledgments:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [keyboard event][] for which all is true: 
+- the event's attribute `key` is a [printable character][] key; and
+- the event's method `getModifierState` returns `false` for any of the [valid modifier keys][]; and
+- the event is [dispatched][] to an [event target][] within a [HTML document][].
+
+## Expectation
+
+For each test target at least one of the following is true:
+ - **Remap**: there is at least one [set of clearly labeled instruments][] to [block events][blocked event] that use the [same key][same key events] as the test target and whose `getModifierState` method returns `false` for any of the [valid modifier keys][]; or
+ - **Focus**: if the [event target][] doesn't have [focus][] the event does not cause [changes in content][] of the [HTML document][].
```

Comment:
And I realise that we're having problems with that… 😱 

---
As far as I understand https://www.w3.org/TR/uievents/#interface-keyboardevent when a key is pressed, the browser is dispatching the keyboard event to the element that has focus and then the element bubbles up (actually it's more complicated) until an ancestor with a listener is found and propagation is stopped.

But that means that the event target **always** has focus when the event is dispatched. In our cases, where the event listener is registered on the `body` element, if focus is on an unrelated element (e.g. the "disable" button), then the event will be dispatched to that event target, which has focus, bubbles up to the `body`, and makes changes in the content. But the event target having focus means that the **Focus** condition passes no matter what 😭 

I do not know how to fix that… We kind of have a notion of "target" which is not the event target (in the examples, the "target" is the input field). But the event listener can be on any ancestor of that "target".

Reply:
You are right. And this made my head hurt!!! I think I came up with a solution. Please do check for flaws, as there might be some.
The SC only looks at the focus condition when the keyboard shortcut is for user interface components (it's not totally clear what this means). But I'm assuming the shortcut should have an effect only when the focus is on a user interface component. Given that the event target will always be the element that has the focus, I'm changing the expectation to check that changes don't happen if the event target is not a widget.